### PR TITLE
Interactivity API: Move Core implementation to compat 6.5 folder

### PR DESCRIPTION
**Pull request by DAreRodz**
## What?

Updates all the Interactivity API PHP files implementation in Gutenberg using the current one in WordPress core (implemented in https://github.com/WordPress/wordpress-develop/pull/5953).

All the related PHPUnit files are not needed anymore in Gutenberg, as there are tests in WordPress core for all features.

### CODE REVIEW

File: lib/experimental/interactivity-api.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 function gutenberg_interactivity_override_script_module_urls( $url ) {
```
Comment (youknowriad) [inline lib/experimental/interactivity-api.php:14]: Why is this file still in experimental? Is it something that will remain in Gutenberg forever? Should it move to compat folder.

Comment (cbravobernal) [reply lib/experimental/interactivity-api.php:14]: This PR will remove that file and move to the external folder:
https://github.com/WordPress/gutenberg/pull/58830

**Review by youknowriad — APPROVED** [review]

**Comment by youknowriad** [comment]
Looks like there's still some e2e test failures.

**Comment by luisherranz** [comment]
They don't look related to the Interactivity API.

**Comment by getdave** [comment]
e2e test failures do indeed look like problems with Synced Pattern work which seems unrelated. Can we force merge this PR to unblock the PHP unit tests?

**Comment by getdave** [comment]
Actually I noticed the e2e tests seem to fail on REST API errors. Is there some conflict here causing REST to fail?

**Comment by youknowriad** [comment]
Ok, so I forced merged this to fix the php unit test but let's continue on the e2e tests, they seem block bindings related.

**Pull request merged**